### DISCUSSION TITLE
strange angle conversions

### DISCUSSION THREAD

**Discussion by neinecke** [esa/SpOC2#18] [Morphing Rovers Q&A] [answered]
Dear SpOC organizers,

I was wondering why the angles are converted back and forth in different fashions. Before giving the current angle to the controller you divide by pi and divide by 2. Before adding the angle_diff output of the controller to the angle you multiply it with pi and divide it by 4. I can't make no sense if it. Why make the live extra difficult for the controller?

Best,
Nils

**Comment by dodo47** [comment, marked as answer]
Hi Nils,

thanks for your question. The reasons for the two examples you are giving are rather different in nature:

1) We rescale the inputs to the neural network, so that every feature has approximately the same value range. Thus, the angle is rescaled by a factor 1/(2pi), as the angle is given in radian. 
2) This might have been explained a bit unclearly in the problem description, sorry! The neural network returns a value between -1 and 1, which is multiplied by the maximally possible angular velocity (which is pi/4, taken from the rover's manual ;) ).

So in both cases, input and output are given as multiples (or rather: fraction) of the absolute maximum value.

I hope this clarifies your question!

Best,
Dominik

**Reply by neinecke** [reply]
Hi Dominik,

Thanks, for the quick reply and clarification.


Best,
Nils
